src/Injections/ICameraHandler.cs

```diff
@@ -4,40 +4,50 @@
 {
     [HarmonyPatch(typeof(CameraController))]
-    [HarmonyPatch("UpdateTransform")]
+    [HarmonyPatch("LateUpdate")]
     public class ICameraUpdateCurrentPosition
     {
         private static Vector3 playerCameraPosition_last;
         private static Quaternion playerCameraRotation_last;
 
         public static void Postfix(CameraController __instance, Camera ___m_camera)
         {
-            // Get camera rotation, angle and position
-            Transform transform = ___m_camera.transform;
-            Quaternion _rotation = transform.rotation;
-            Vector3 _position = transform.position;
-
-            // Check if the camera moved
-            if (Vector3.Distance(_position, playerCameraPosition_last) > 1 || playerCameraRotation_last != _rotation)
+            // Make sure the camera object is defined, this should always happen but for some strange reason it didnt (look bellow)
+            // This is not happening if i test it localy. (debug using AllocConsole in CSM.cs)
+            // https://github.com/DominicMaas/Tango/issues/155
+            if (___m_camera && MultiplayerManager.Instance.PlayerList.Count > 0)
             {
-                // Store camera rotation and position
-                playerCameraPosition_last = _position;
-                playerCameraRotation_last = _rotation;
+                // Get camera rotation, angle and position
+                Transform transform = ___m_camera.transform;
+                Quaternion _rotation = transform.rotation;
+                Vector3 _position = transform.position;
 
-                // Send info to all clients
-                Command.SendToAll(new PlayerLocationCommand
+                // Update the player camera position every time the camera moves OR rotates
+                // This should not be that demanding on the server (might be demanding if you are playing with a lot of players
+                // We could also make the server update the player locations every x seconds by storing all the info on the server and send it to all clients every 1 second?
+                if (Vector3.Distance(_position, playerCameraPosition_last) > 1 || playerCameraRotation_last != _rotation)
                 {
-                    PlayerName = MultiplayerManager.Instance.CurrentClient.Config.Username,
-                    PlayerCameraPosition = _position,
-                    PlayerCameraRotation = _rotation,
-                    PlayerCameraHeight = __instance.m_currentHeight,
-                    PlayerColor = JoinGamePanel.playerColor
-                });
+                    // Store camera rotation and position
+                    playerCameraPosition_last = _position;
+                    playerCameraRotation_last = _rotation;
+
+                    // Send info to all clients
+                    Command.SendToAll(new PlayerLocationCommand
+                    {
+                        PlayerName = MultiplayerManager.Instance.CurrentClient.Config.Username,
```

Comment:
This causes an exception on the server (where the client Config is not set). You need to get the name differently when the game acts as the server (`MultiplayerManager.Instance.CurrentRole == MultiplayerRole.Server`) by using `MutliplayerManager.Instance.CurrentServer.Config.Username`.

Reply:
Fixed in latest commit